legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java

```diff
@@ -124,6 +124,107 @@ private static DataSpaceExecutableResult buildExecutableResult(ResultType result
         return null;
     }
 
+    private static List<DataSpaceExecutableAnalysisInfo> buildDataSpaceExecutableAnalysisInfo(PureModel pureModel, DataSpace dataSpaceProtocol, PureModelContextData pureModelContextData)
+    {
+        List<DataSpaceExecutableAnalysisInfo> dataSpaceExecutableAnalysisInfos = null;
+        Set<String> qualifiedMappingPath = dataSpaceProtocol.executionContexts.stream().map(e -> e.mapping.path).collect(Collectors.toSet());
+        Set<String> qualifiedRuntimePath = dataSpaceProtocol.executionContexts.stream().map(e -> e.defaultRuntime.path).collect(Collectors.toSet());
+        if (dataSpaceProtocol.executables != null)
+        {
+            dataSpaceExecutableAnalysisInfos = dataSpaceProtocol.executables.stream()
+                    .filter(ex -> ex instanceof DataSpacePackageableElementExecutable)
+                    .map(executable ->
+                    {
+                        String path = ((DataSpacePackageableElementExecutable) executable).executable.path;
+                        try
```

Comment:
dont understand the need for the try/catch herem can't we just check the type of `PackageableElement`.

Reply:
I made a mistake when resolving your comment "I think we are in the right direction, but we need to be mindful of not restoring the same exact information. We should store only things we need. I think we can keep the template info inside execution context based on your filtering. But for functions, it should be stored globally in the DataSpaceAnalysisResult class.".


I thought you wanted to keep the inline one as what it is instead of writing an info for it. 

I have added the inline one back. pleas recheck